pkg/environment/environmentApply.go

```diff
@@ -173,13 +176,31 @@ func (a *Apply) Destroy() error {
 		return err
 	}
 	if isMerged {
-		changedNamespaces, err := a.nsCreateRawChangedFilesInPR(a.Options.ClusterCtx, a.Options.PRNumber)
+		changedNamespaces, err := a.nsCreateRawChangedFilesInPR(a.Options.ClusterDir, a.Options.PRNumber)
 		fmt.Println("Namespaces changed in PR", changedNamespaces)
 		if err != nil {
 			return err
 		}
+		kubeClient, err := authenticate.CreateClientFromConfigFile(a.Options.KubecfgPath, a.Options.ClusterCtx)
+		if err != nil {
+			return err
+		}
+
+		// GetAllNamespacesFromCluster
+		namespaces, err := namespace.GetAllNamespacesFromCluster(kubeClient)
+		if err != nil {
+			return err
+		}
+
 		for _, namespace := range changedNamespaces {
 			a.Options.Namespace = namespace
+			if a.Options.SkipProdDestroy {
+				if isProductionNs(namespace, namespaces) {
```

Comment:
Why is this a nested if statement? Could we get rid of the nested if and use:

```
if a.Options.SkipProdDestroy && isProductionNs(namespace, namespaces) {
```

Reply:
Thanks @jaskaransarkaria . I might have been overthinking. Fixed in the later commit